x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/edit_connector_flyout/index.tsx

```diff
@@ -146,83 +158,82 @@ const EditConnectorFlyoutComponent: React.FC<EditConnectorFlyoutProps> = ({
       setTestExecutionResult(none);
     },
     [setIsFormModified]
   );
 
   const closeFlyout = useCallback(() => {
+    if (isFormModified) {
+      setShowConfirmModal(true);
+      return;
+    }
     onClose();
-  }, [onClose]);
+  }, [onClose, isFormModified, setShowConfirmModal]);
 
-  const onClickSave = useCallback(
-    async (closeAfterSave: boolean = true) => {
-      setPreSubmitValidationErrorMessage(null);
+  const onClickSave = useCallback(async () => {
+    setPreSubmitValidationErrorMessage(null);
 
-      const { isValid, data } = await submit();
-      if (!isMounted.current) {
-        // User has closed the flyout meanwhile submitting the form
-        return;
-      }
+    const { isValid, data } = await submit();
+    if (!isMounted.current) {
+      // User has closed the flyout meanwhile submitting the form
+      return;
+    }
 
-      if (isValid) {
-        if (preSubmitValidator) {
-          const validatorRes = await preSubmitValidator();
+    if (isValid) {
+      if (preSubmitValidator) {
+        const validatorRes = await preSubmitValidator();
 
-          if (validatorRes) {
-            setPreSubmitValidationErrorMessage(validatorRes.message);
-            return;
-          }
+        if (validatorRes) {
+          setPreSubmitValidationErrorMessage(validatorRes.message);
+          return;
         }
+      }
+
+      /**
+       * At this point the form is valid
+       * and there are no pre submit error messages.
+       */
+
+      const { name, config, secrets } = data;
+      const validConnector = {
+        id: connector.id,
+        name: name ?? '',
+        config: config ?? {},
+        secrets: secrets ?? {},
+      };
 
+      const updatedConnector = await updateConnector(validConnector);
+
+      if (updatedConnector) {
         /**
-         * At this point the form is valid
-         * and there are no pre submit error messages.
+         * ConnectorFormSchema has been saved.
+         * Set the from to clean state.
          */
+        onFormModifiedChange(false);
 
-        const { name, config, secrets } = data;
-        const validConnector = {
-          id: connector.id,
-          name: name ?? '',
-          config: config ?? {},
-          secrets: secrets ?? {},
-        };
-
-        const updatedConnector = await updateConnector(validConnector);
-
-        if (updatedConnector) {
-          /**
-           * ConnectorFormSchema has been saved.
-           * Set the from to clean state.
-           */
-          onFormModifiedChange(false);
-
-          if (onConnectorUpdated && updatedConnector) {
-            onConnectorUpdated(updatedConnector);
-          }
-
-          if (closeAfterSave) {
-            closeFlyout();
-          }
+        if (onConnectorUpdated) {
+          onConnectorUpdated(updatedConnector);
         }
-
-        return updatedConnector;
+        setIsSaved(true);
+        setIsEdit(false);
+        setIsEdit(true);
```

Comment:
After the first "Save" isFormModified hook does not work. It keeps using the first values that were passed to the form. 
I used this trick (setEdit true/false) to reset form state, so it's forced to use the latest connector data.